compiler/rustc_passes/src/check_attr.rs

```diff
@@ -717,6 +717,39 @@ impl CheckAttrVisitor<'tcx> {
         true
     }
 
+    /// Checks that `doc(test(...)` attribute contains only valid attributes. Returns `true` if
+    /// valid.
+    fn check_test_attr(&self, meta: &NestedMetaItem) -> bool {
+        let mut is_valid = true;
+        if let Some(metas) = meta.meta_item_list() {
+            for i_meta in metas {
+                match i_meta.name_or_empty() {
+                    sym::attr | sym::no_crate_inject => {}
+                    name => {
+                        self.tcx
+                            .sess
+                            .struct_span_err(
+                                i_meta.span(),
+                                &format!("unknown `doc(test)` attribute `{}`", name,),
```

Comment:
Could you change this to

```suggestion
                                &format!("unknown `doc(test)` attribute `{}`", rustc_ast_pretty::pprust::path_to_string(&i_meta.path)),
```

like other places where we emit the lint so that things like `doc(test(foo::bar))` don't show up as ```unknown `doc(test)` attribute `` ```?

Reply:
:+1: